src/main/java/uk/gov/ons/ctp/response/notify/util/BitCalculator.java

```diff
@@ -0,0 +1,207 @@
+package uk.gov.ons.ctp.response.notify.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.math.BigInteger;
+
+/**
+ * This is a simple class to take a gov.notify key or other arbitrary string representing a number and a radix and
+ * return the number of bits of data in the key. This is primarily for working out the validity of notify keys.
+ * @author innesm
+ */
+public class BitCalculator {
```

Comment:
Is there not a library for validating the gov notify api keys. You'd think that they were validating it themselves? If there is another library out there I guess it would be preferable to use that

Reply:
Nope. This is the Jose JWT library that's validating the key, gov notify has nothing to do with it.  I googled and googled for something that would do it and all I found was an algorithm on SO that I have implemented here